tests/integration/security/testdata/beta-mtls-off-headless-nake.yaml

```diff
@@ -0,0 +1,20 @@
+# Alpha API disable, making test more hermetic.
+# No DR is added for this test. enableAutoMtls is expected on by default.
+# Workload-level PeerAuthentication will expectedly cause trouble to connect to the corresponding service
+apiVersion: authentication.istio.io/v1alpha1
+kind: MeshPolicy
+metadata:
+  name: "default"
+  annotations:
+    test-suite: "beta-mtls-off-headless-nake"
+spec: {}
+---
+apiVersion: "security.istio.io/v1beta1"
+kind: "PeerAuthentication"
+metadata:
+  name: "default"
+  annotations:
+    test-suite: "beta-mtls-off-headless-nake"
+spec:
+  mtls:
+    mode: DISABLE  # This is to simulate a headless service without proxy
```

Comment:
this is different from the actual naked headless, and involve different code path.

you can totally use echo app's configuration adding annotations and make it headless service.

Reply:
The tricky part is that `CreateContext` is called by many tests. And there are many failures once i add a new headless naked service.

I will try again.